src/Frame.h

```diff
@@ -0,0 +1,35 @@
+#ifndef __FRAME_H__
+#define __FRAME_H__
+
+#include <string>
+#include <cstdint>
+
+using namespace std;
+
+class Frame {
+public:
+    // Getters
```

Comment:
I think it will simplify things...especially since he wants things threaded again. All those getters and setters will have to be made thread safe.